scripts/config.py

```diff
@@ -513,29 +517,36 @@ def _config_CUDA(self):
                 # in case user specified some wrapper of the compiler
                 # get rid of it: 'ccache g++' -> 'g++'
                 currentCxx = self.get(Config.KEY_CXX).split()[-1]
-                cxxVersion, cxxStrVersion = self._get_GCC_version(currentCxx)
                 nvccVersion, nvccFullVersion = self._get_CUDA_version(
                     self.configDict["NVCC"])
                 if self.configDict["CXX_CUDA"] == '':
+                    self.outOfUsrBinGCC = False
                     print(
                         yellow('Checking for compatible GCC to be used with your CUDA'))
                     candidates = self._get_compatible_GCC(nvccVersion)
                     for c in candidates:
-                        p = 'g++-' + c
-                        if checkProgram(p, False):
-                            self.configDict["CXX_CUDA"] = p
-                            break
+                        if c == '':
```

Comment:
This is very cluttered and it's not obvious why you have to do that if you have compiler in non-standard location.